gateway-service/src/main/java/com/ca/mfaas/gateway/security/handler/UnauthorizedHandler.java

```diff
@@ -41,10 +38,12 @@ public UnauthorizedHandler(ErrorService errorService, ObjectMapper objectMapper)
     @Override
     public void commence(HttpServletRequest request, HttpServletResponse response, AuthenticationException authException) throws IOException {
         log.debug("Unauthorized access to '{}' endpoint", request.getRequestURI());
-        ApiMessage message = errorService.createApiMessage("com.ca.mfaas.security.authenticationRequired", request.getRequestURI());
 
         response.setStatus(HttpStatus.UNAUTHORIZED.value());
-        response.setContentType(MediaType.APPLICATION_JSON_UTF8_VALUE);
-        mapper.writeValue(response.getWriter(), message);
+        response.addHeader("WWW-Authenticate", ApimConstants.BASIC_AUTHENTICATION_PREFIX);
+
+        ApiMessage message = errorService.createApiMessage("com.ca.mfaas.gateway.security.invalidCredentials", request.getRequestURI());
+        PrintWriter writer = response.getWriter();
+        writer.println("HTTP Status 401 : " + message.toString());
```

Comment:
The solution is good. Just we need to keep the error message as parsable. like JSON. So I would suggest printing message in application/json as previous.